charts/orchestrator/templates/tekton-pipeline.yaml

```diff
@@ -0,0 +1,164 @@
+            ssh-add "${PARAM_USER_HOME}"/.ssh/id_rsa
+            git clone $(params.gitConfigUrl) workflow-config
+    - name: flatten-workflow
+      runAfter: ["fetch-workflow"]
+      taskRef:
+        name: flattener
+      workspaces:
+        - name: workflow-source
+          workspace: workflow-source
+      params:
+        - name: workflowId
+          value: $(params.workflowId)
+        - name: convertToFlat
+          value: $(params.convertToFlat)
+    - name: build-manifests
+      runAfter: ["flatten-workflow"]
+      taskRef:
+        name: build-manifests
+      workspaces:
+        - name: workflow-source
+          workspace: workflow-source
+      params:
+        - name: workflowId
+          value: $(params.workflowId)
+    - name: build-config
+      runAfter: ["build-manifests", "fetch-workflow-config"]
+      taskRef:
+        name: build-config
+      workspaces:
+        - name: workflow-source
+          workspace: workflow-source
+        - name: workflow-config
+          workspace: workflow-config
+      params:
+        - name: workflowId
+          value: $(params.workflowId)
+        - name: imageTag
+          value: $(tasks.fetch-workflow.results.commit)
+    - name: build-and-push-image
+      runAfter: ["flatten-workflow"]
+      taskRef:
+        name: buildah
+        kind: ClusterTask        
+      workspaces:
+        - name: source
+          workspace: workflow-source
+        - name: dockerconfig
+          workspace: docker-credentials
+      params:
+        - name: IMAGE
+          value: quay.io/orchestrator/serverless-workflow-$(params.workflowId):$(tasks.fetch-workflow.results.commit)
+        - name: DOCKERFILE
+          value: flat/workflow-builder.Dockerfile
+        - name: CONTEXT
+          value: flat/$(params.workflowId)
+        - name: BUILD_EXTRA_ARGS
+          value: --ulimit nofile=4096:4096             
+    - name: push-workflow-config
+      runAfter: ["build-config", "build-and-push-image"]
+      taskRef:
+        name: git-cli
+      workspaces:
+        - name: source
+          workspace: workflow-config
+        - name: ssh-directory
+          workspace: ssh-creds
+      params:
+        - name: GIT_USER_NAME
+          value: Parodos Tekton Pipeline
+        - name: GIT_USER_EMAIL
+          value: parodos@redhat.com
+        - name: USER_HOME
+          value: /home/git
+        - name: GIT_SCRIPT
+          value: |
+            WORKFLOW_COMMIT=$(tasks.fetch-workflow.results.commit)
+            echo "WORKFLOW_COMMIT is $WORKFLOW_COMMIT"
+
+            eval "$(ssh-agent -s)"
+            ssh-add "${PARAM_USER_HOME}"/.ssh/id_rsa
+
+            # git config --global --add safe.directory /workspace/source/workflow
+            # git clone $(params.gitUrl) workflow
+            cd workflow-config
+            git add .
+            echo "--git status"
+            git status
+            echo "--git diff"
+            git diff
+            # git config --global --list
```

Comment:
needed?

Reply:
I would keep the `git diff` part for debugging purposes at least

Reply:
makes sensd